File: code/samsung_sw_test/14888.py
import sys
input = lambda : sys.stdin.readline().rstrip()

def calc(n1, n2, oper):
    if oper==0:
        return n1 + n2
    elif oper == 1:
        return n1 - n2
    elif oper == 2:
        return n1 * n2
    elif oper == 3:
        if n1 < 0 and n2 > 0:
            return -((-n1) // n2)
        return n1 // n2

def recur(cur, total):
    global max_result, min_result
    if cur == N:
        min_result = min(min_result, total)
        max_result = max(max_result, total)
        return
    for i in range(4):
        if oper[i]:
            oper[i] -= 1
            recur(cur+1, calc(total, numbers[cur], i))
            oper[i] += 1
    

max_result = -1000000001
min_result = 1000000001

N = int(input())
numbers = list(map(int, input().split()))
oper = list(map(int, input().split()))
recur(1, numbers[0])
print(max_result)
print(min_result)

File: samsung_sw_test/14888.cpp
#include <iostream>

using namespace std;

int oper[4];
int min_result = 1000000001;
int max_result = -1000000001;
int numbers[12];

int calc(int n1, int n2, int oper) {
  if (oper==0) return n1 + n2;
  else if (oper==1) return n1 - n2;
  else if (oper==2) return n1 * n2;
  else if (oper==3) return n1 / n2;
  else return 100000;
}

void recur(int cur, int total, int N) {
  // cout << cur << " " << total << " " << N << endl;
  if (cur == N) {
    if (min_result > total) min_result = total;
    if (max_result < total) max_result = total;
    return;
  }
  for (int i=0; i<4; i++) {
    if (oper[i] > 0){
      oper[i] -= 1;
      recur(cur+1, calc(total, numbers[cur], i), N);
      oper[i] += 1;
    }
  }
}



int main() {
  int N;
  int i;

  cin >> N;
  for (i=0; i<N; i++) {
    cin >> numbers[i];
  }
  for (i=0; i<4; i++) {
    cin >> oper[i];
  }
  recur(1, numbers[0], N);
  cout << max_result << "\n" << min_result;
  return 0;
}